src/Dapr.Actors/Runtime/DaprStateProvider.cs

```diff
@@ -77,10 +80,60 @@ public async Task<bool> ContainsStateAsync(string actorType, string actorId, str
 
         public async Task SaveStateAsync(string actorType, string actorId, IReadOnlyCollection<ActorStateChange> stateChanges, CancellationToken cancellationToken = default)
         {
-            await this.DoStateChangesTransactionallyAsync(actorType, actorId, stateChanges, cancellationToken);
+            if (this.daprInteractor.GetType().Name.Equals("DaprGrpcInteractor")){
+                await this.DoStateChangesTransactionallyAsyncGrpc(actorType, actorId, stateChanges, cancellationToken);
+            } else {
+                await this.DoStateChangesTransactionallyAsyncHttp(actorType, actorId, stateChanges, cancellationToken);
+            }
```

Comment:
The interface of the interactor makes this unnecessary. Whatever the underlying object is, we'll use its method. So no need to separate them like this.